File: tags/v1_3_5/JavaYacas/net/sf/yacas/PatternClass.java
package net.sf.yacas;


/// Wrapper for YacasPatternPredicateBase.
/// This class allows a YacasPatternPredicateBase to be put in a
/// LispGenericObject.
class PatternClass extends GenericClass
{
  public PatternClass(YacasPatternPredicateBase aPatternMatcher)
  {
    iPatternMatcher = aPatternMatcher;
  }

  public boolean Matches(LispEnvironment  aEnvironment, LispPtr aArguments) throws Exception
  {
    LispError.LISPASSERT(iPatternMatcher != null);
    boolean result;
    result = iPatternMatcher.Matches(aEnvironment, aArguments);
    return result;
  }
  public boolean Matches(LispEnvironment  aEnvironment, LispPtr[] aArguments) throws Exception
  {
    LispError.LISPASSERT(iPatternMatcher != null);
    boolean result;
    result = iPatternMatcher.Matches(aEnvironment, aArguments);
    return result;
  }
  //From GenericClass
  public String Send(LispArgList aArgList)
  {
    return null;
  }
  public String TypeName()
  {
      return "\"Pattern\"";
  }

  protected YacasPatternPredicateBase iPatternMatcher;
}

File: cyacas/libyacas/src/patternclass.cpp
#include "yacas/patternclass.h"

PatternClass::PatternClass(YacasPatternPredicateBase* aPatternMatcher)
: iPatternMatcher(aPatternMatcher)
{
}

PatternClass::~PatternClass()
{
    delete iPatternMatcher;
}

const char* PatternClass::TypeName() const
{
    return "\"Pattern\"";
}

bool PatternClass::Matches(LispEnvironment& aEnvironment,
                                  LispPtr& aArguments)
{
    assert(iPatternMatcher);
    return iPatternMatcher->Matches(aEnvironment, aArguments);
}

bool PatternClass::Matches(LispEnvironment& aEnvironment,
                                  LispPtr* aArguments)
{
    assert(iPatternMatcher);
    return iPatternMatcher->Matches(aEnvironment, aArguments);
}
